website/src/docs/core.md

```diff
@@ -946,6 +946,23 @@ uppy.on('upload-error', (file, error, response) => {
 })
 ```
 
+### `upload-stalled`
+
+Fired when an upload is seemingly stalled. Use this event to display a message on the UI to tell the user they might want to retry the upload.
+
+```js
+uppy.on('upload-stalled', (error, files) => {
+  console.log(error, files)
+  const uploadProgressHandler = (file) => {
+    if (files.includes(file)) {
+      uppy.info('The upload is no longer stalled')
+      uppy.off('upload-progress', uploadProgressHandler)
+    }
+  }
+  uppy.on('upload-progress', uploadProgressHandler)
```

Comment:
Why do we need to set this handler again?

Reply:
It's to illustrate how to detect when the upload is no longer stalled.

Reply:
Perhaps we can add a comment about it. Or actually show an example that informs the user, as that's what we talk about above 

> "Use this event to display a message on the UI to tell the user they might want to retry the upload."

Now it's a bit confusing to me how retrying works together with this.